app/cdap/components/DataPrep/Directives/Calculate/index.js

```diff
@@ -44,19 +44,22 @@ export default class Calculate extends Component {
     super(props);
     this.VALID_TYPES = NUMBER_TYPES.concat(['string']);
 
-    const {isDisabled, crossColumn, columns, columnType} = this.parseColumns();
+    const {isDisabled, crossColumn, multiColumn, columns, columnTypes, areColsMixedTypes} = this.parseColumns();
 
     this.isDisabled = isDisabled;
     this.crossColumn = crossColumn;
+    this.multiColumn = multiColumn;
     this.columns = columns;
-    this.columnType = columnType;
+    this.columnTypes = columnTypes;
+    this.areColsMixedTypes = areColsMixedTypes;
 
     this.defaultState = {
       operationPopoverOpen: null,
       operationInput: 1,
       createNewColumn: false,
       newColumnInput: this.props.column + T.translate(`${COPY_NEW_COLUMN_PREFIX}.inputSuffix`),
-      isDisabled: this.VALID_TYPES.indexOf(this.columnType) === -1 || this.isDisabled,
+      isDisabled: !this.columnTypes.every(type => this.VALID_TYPES.includes(type)) // Check all columnTypes valid
```

Comment:
I don't think the comment is necessary - if the function of the code can be easily inferred, you don't need to comment. If you do want to keep the comment, make sure it doesn't split up the expression.